### PR TITLE
fix(rum): Compute the true failure rate for web vitals

### PR DESCRIPTION
**Pull request by Zylphrex**
Currently the failure rate for web vitals is calculated using only what's
visible on the histograms. This was confusing because when a zoom happens the
failure rate will change with it. This change queries for the actual failure
rate based on the query conditions independent of the visible histogram.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -1743,6 +1743,16 @@ def validate_argument_count(self, field, arguments):
             aggregate=["count", None, None],
             result_type="integer",
         ),
+        Function(
+            "count_geq",
```
Comment (markstory) [inline src/sentry/api/event_search.py:1747]: Are we planning on exposing this to customers as well? We might need a more clear name if we do.

File: src/sentry/static/sentry/app/views/performance/realUserMonitoring/transactionVitals.tsx [hunk 2]
```diff
@@ -53,21 +53,31 @@ class TransactionVitals extends React.Component<Props> {
                 {results => (
                   <React.Fragment>
                     {vitals.map((vital, index) => {
+                      const details = WEB_VITAL_DETAILS[vital];
                       const error =
                         summaryResults.error !== null || results.error !== null;
-                      const alias = getAggregateAlias(
+                      const percentile_alias = getAggregateAlias(
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/realUserMonitoring/transactionVitals.tsx:59]: typescript variable names are generally camelBacked.

File: src/sentry/static/sentry/app/views/performance/realUserMonitoring/transactionVitals.tsx [hunk 3]
```diff
@@ -53,21 +53,31 @@ class TransactionVitals extends React.Component<Props> {
...
+                      const count_alias = getAggregateAlias(`count_geq(${vital}, 0)`);
+                      const failed_alias = getAggregateAlias(
+                        `count_geq(${vital}, ${details.failureThreshold})`
+                      );
+                      const data = summaryResults.tableData?.data?.[0];
+                      const summary = (data?.[percentile_alias] ?? null) as number | null;
+                      const numerator = (data?.[failed_alias] ?? 0) as number;
+                      const denominator = (data?.[count_alias] ?? 0) as number;
+                      const failureRate =
+                        denominator <= 0 ? null : numerator / denominator;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/realUserMonitoring/transactionVitals.tsx:71]: If you coalesced to 0 here you would have a simpler type in `VitalCard` and not need to handle nulls there.

File: tests/snuba/api/endpoints/test_organization_events_v2.py [hunk 4]
```diff
@@ -2477,6 +2477,21 @@ def test_limit_number_of_fields(self):
...
 
+    def test_count_geq_query(self):
+        self.store_event(self.transaction_data, self.project.id)
+
+        response = self.do_request({"field": "count_geq(measurements.fcp, {})".format(0)})
+        assert response.status_code == 200
+        assert len(response.data["data"]) == 1
+        assert response.data["data"][0]["count_geq_measurements_fcp_0"] == 1
+
+        # an value that's a little bigger than the stored fcp
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_events_v2.py:2488]:
```suggestion
        # a value that's a little bigger than the stored fcp
```

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/event_search.py:1747]: That is a clearer name, but really long. I don't have productive suggestions for another name though.

Comment (Zylphrex) [reply src/sentry/api/event_search.py]: Changed the name to `count_at_least`

**Review by wmak — APPROVED** [review]
Nice

**Review by markstory — APPROVED** [review]

**Pull request merged**